x-pack/plugins/security_solution/public/timelines/store/middlewares/timeline_note.ts

```diff
@@ -79,6 +88,36 @@ export const addNoteToTimelineMiddleware: (kibana: CoreStart) => Middleware<{},
             },
           })
         );
+
+        // In case a note was added to an unsaved timeline, we need to make sure to update the timeline
+        // locally and then remotely again in order not to lose the SO associations.
+        // This also involves setting the status and the default title.
+        if (!timeline.savedObjectId && response.note.timelineId && response.note.timelineVersion) {
+          const currentTimeline = selectTimelineById(store.getState(), localTimelineId);
+          await store.dispatch(
+            updateTimeline({
+              id: localTimelineId,
+              timeline: {
+                ...currentTimeline,
+                savedObjectId: response.note.timelineId,
+                version: response.note.timelineVersion || currentTimeline.version,
+                status: TimelineStatus.active,
+                title: currentTimeline.title || UNTITLED_TIMELINE,
+              },
+            })
+          );
+          await store.dispatch(saveTimeline({ id: localTimelineId, saveAsNew: false }));
+        }
+
+        // Events can be automatically pinned
+        if (isAddNoteToEventAction(action) && action.payload.pinEvent) {
+          await store.dispatch(
+            pinEvent({
+              id: localTimelineId,
+              eventId: action.payload.eventId,
+            })
+          );
+        }
```

Comment:
I have no historical context on this feature but to me it feels like the `pinEvent` option is not needed and we want to always pin an event when a comment has been added to it. Wdyt @michaelolo24 ?

Reply:
So I wasn't around when this decision was made, but yes, I agree. If a user made a comment on any kind of event, they would want to quickly reference it later on, so it would make logical sense to have it auto-pinned.